# Análisis de datos con Pyhon - Pre-entrega
Nombre y apellido: Franco Verónica

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


# Etapa 1

## Script básico
Actividades 2 y 3

In [3]:
# Generamos una lista de diccionarios que contenga algunos datos
ventas = [
  {'producto': 'notebook', 'precio': 1500.0, 'cantidad': 3},
  {'producto': 'monitor', 'precio': 500.0, 'cantidad': 5},
  {'producto': 'mouse', 'precio': 20.0, 'cantidad': 15}
]

In [4]:
# Calculamos las ventas acumuladas - map
venta_acumuladas_map = sum(map(lambda x: x["precio"] * x["cantidad"], ventas))
print(venta_acumuladas_map)

7300.0


## Import Datasets
Actividad 1

In [5]:
import pandas as pd

In [7]:
# Import dataset ventas
df_ventas = pd.read_excel("https://github.com/Vero-Franco/SynthData_pre-entrega/blob/main/ventas.xlsx?raw=true")

In [9]:
# Import dataset clientes
df_clientes = pd.read_excel("https://github.com/Vero-Franco/SynthData_pre-entrega/blob/main/clientes.xlsx?raw=true")

In [11]:
# Import dataset marketing
df_marketing = pd.read_excel("https://github.com/Vero-Franco/SynthData_pre-entrega/blob/main/marketing.xlsx?raw=true")

## Dataframe ventas

### Análisis exploratorio de **datos**
Actividad 4

Aplicamos los métodos de EDA vistos en clase, `info`, `columns`, `dtypes`, `head`, `tail`, `sample`, `unique`, `value_counts`, `describe`, `duplicated`, `isnull`, etc. y documentamos las observaciones:

In [12]:
# Aplicamos info() para familiarizarnos con el dataframe
df_ventas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3035 entries, 0 to 3034
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   id_venta     3035 non-null   int64         
 1   producto     3035 non-null   object        
 2   precio       3033 non-null   float64       
 3   cantidad     3033 non-null   float64       
 4   fecha_venta  3035 non-null   datetime64[ns]
 5   categoria    3035 non-null   object        
dtypes: datetime64[ns](1), float64(2), int64(1), object(2)
memory usage: 142.4+ KB


In [13]:
# Visualizamos los primeros registros con head()
# también podemos usar tail() para visualizar los últimos
df_ventas.head()

id_venta           producto  precio  cantidad fecha_venta  \
0       792  Cuadro decorativo   69.94       5.0  2024-01-02   
1       811    Lámpara de mesa  105.10       5.0  2024-01-02   
2      1156           Secadora   97.96       3.0  2024-01-02   
3      1372           Heladera  114.35       8.0  2024-01-02   
4      1546           Secadora  106.21       4.0  2024-01-02   

           categoria  
0         Decoración  
1         Decoración  
2  Electrodomésticos  
3  Electrodomésticos  
4  Electrodomésticos

In [14]:
# Usar sample() para tomar registros al azar
df_ventas.sample(5)

id_venta             producto  precio  cantidad fecha_venta  \
297       2851  Parlantes Bluetooth  110.42       2.0  2024-02-03   
2321      1187      Horno eléctrico   99.57       8.0  2024-09-30   
1389      2126               Laptop  107.26       4.0  2024-06-02   
1809      1375             Lavadora   78.13       5.0  2024-07-29   
2274      2248           SmartWatch   88.46       8.0  2024-09-25   

              categoria  
297         Electrónica  
2321  Electrodomésticos  
1389        Electrónica  
1809  Electrodomésticos  
2274        Electrónica

In [15]:
# Analizamos el campo precio
# Si es numericos se puede aplicar describe()
df_ventas["precio"].sample(5)

942     32.00
1049    88.02
2770    97.60
231     30.31
2868    46.02
Name: precio, dtype: float64

In [16]:
# Analizamos el campo cantidad
# si es numerico se puede aplicar describe()
df_ventas["cantidad"].sample(5)

2506     6.0
551      1.0
1567     2.0
753     10.0
538      7.0
Name: cantidad, dtype: float64

In [17]:
# Analizamos el campo fecha_venta
df_ventas["fecha_venta"].sample(5)

270    2024-02-01
2101   2024-09-02
1184   2024-05-10
354    2024-02-10
641    2024-03-15
Name: fecha_venta, dtype: datetime64[ns]

In [18]:
# Analizamos el campo categoria
# Probar con value_counts y unique
df_ventas["categoria"].value_counts()

categoria
Decoración           1015
Electrónica          1012
Electrodomésticos    1008
Name: count, dtype: int64

### Calidad de datos
Actividad 5

In [19]:
# Buscamos registros duplicados
df_ventas.duplicated().sum()

np.int64(35)

In [20]:
# Visualizamos los duplicados
# keep: 'first', 'last', False
df_ventas[df_ventas.duplicated(keep="first")]

id_venta                producto  precio  cantidad fecha_venta  \
826        56                Cortinas   66.24       5.0  2024-04-05   
827       421         Lámpara de mesa  114.83       9.0  2024-04-05   
828       424       Jarrón decorativo   87.94       2.0  2024-04-05   
829      1868                Cafetera   62.23       1.0  2024-04-05   
830      2545             Auriculares   32.81      11.0  2024-04-05   
831      2778              SmartWatch   70.66       7.0  2024-04-05   
844       145                Alfombra   41.72       8.0  2024-04-06   
845       300         Adorno de pared   67.75       9.0  2024-04-06   
846       439         Lámpara de mesa   35.43       2.0  2024-04-06   
847       906                Cortinas   84.04       3.0  2024-04-06   
848       959         Lámpara de mesa   76.32       7.0  2024-04-06   
849      1376                Secadora   69.58       7.0  2024-04-06   
850      1776                Secadora   33.55       3.0  2024-04-06   
851      1784              Aspiradora  115.24       3.0  2024-04-06   
852      1974              Aspiradora   87.32       7.0  2024-04-06   
853      2080  Consola de videojuegos  100.69       7.0  2024-04-06   
854      2108              SmartWatch   79.37      10.0  2024-04-06   
855      2169               Proyector  120.34       1.0  2024-04-06   
867       213       Espejo decorativo   40.76       8.0  2024-04-07   
868       345                Alfombra   56.78       9.0  2024-04-07   
869       398       Rincón de plantas   77.30       7.0  2024-04-07   
870       509         Lámpara de mesa   72.50       1.0  2024-04-07   
871       821         Lámpara de mesa   71.17      10.0  2024-04-07   
872      1642                Heladera   88.78       2.0  2024-04-07   
873      1956                Secadora   42.59      12.0  2024-04-07   
874      2085             Auriculares  122.56       6.0  2024-04-07   
875      2096                  Laptop   83.25       6.0  2024-04-07   
876      2356                  Laptop   30.16       6.0  2024-04-07   
877      2373              Smartphone  123.60      12.0  2024-04-07   
884       204       Jarrón decorativo   94.75       1.0  2024-04-08   
885      1381      Freidora eléctrica   38.12       2.0  2024-04-08   
886      2365             Auriculares   92.91      11.0  2024-04-08   
887      2486                  Laptop  124.95      11.0  2024-04-08   
888      2506                  Laptop   34.74       1.0  2024-04-08   
889      2705             Auriculares   87.29       2.0  2024-04-08   

             categoria  
826         Decoración  
827         Decoración  
828         Decoración  
829  Electrodomésticos  
830        Electrónica  
831        Electrónica  
844         Decoración  
845         Decoración  
846         Decoración  
847         Decoración  
848         Decoración  
849  Electrodomésticos  
850  Electrodomésticos  
851  Electrodomésticos  
852  Electrodomésticos  
853        Electrónica  
854        Electrónica  
855        Electrónica  
867         Decoración  
868         Decoración  
869         Decoración  
870         Decoración  
871         Decoración  
872  Electrodomésticos  
873  Electrodomésticos  
874        Electrónica  
875        Electrónica  
876        Electrónica  
877        Electrónica  
884         Decoración  
885  Electrodomésticos  
886        Electrónica  
887        Electrónica  
888        Electrónica  
889        Electrónica

In [21]:
df_ventas[df_ventas.duplicated(keep=False)].sort_values(by="producto")

id_venta         producto  precio  cantidad fecha_venta    categoria
845       300  Adorno de pared   67.75       9.0  2024-04-06   Decoración
833       300  Adorno de pared   67.75       9.0  2024-04-06   Decoración
857       345         Alfombra   56.78       9.0  2024-04-07   Decoración
832       145         Alfombra   41.72       8.0  2024-04-06   Decoración
868       345         Alfombra   56.78       9.0  2024-04-07   Decoración
..        ...              ...     ...       ...         ...          ...
854      2108       SmartWatch   79.37      10.0  2024-04-06  Electrónica
831      2778       SmartWatch   70.66       7.0  2024-04-05  Electrónica
825      2778       SmartWatch   70.66       7.0  2024-04-05  Electrónica
877      2373       Smartphone  123.60      12.0  2024-04-07  Electrónica
866      2373       Smartphone  123.60      12.0  2024-04-07  Electrónica

[70 rows x 6 columns]

In [22]:
# Buscamos datos nulos
df_ventas.isnull().sum()

id_venta       0
producto       0
precio         2
cantidad       2
fecha_venta    0
categoria      0
dtype: int64

In [23]:
# Visualizar los nulos
df_ventas.isnull().any(axis=1)

0       False
1       False
2       False
3       False
4       False
        ...  
3030    False
3031    False
3032    False
3033    False
3034    False
Length: 3035, dtype: bool

In [24]:
# muestra cuales son los elementos con valores nulos
df_ventas[df_ventas.isnull().any(axis=1)]

id_venta               producto  precio  cantidad fecha_venta  \
136       627  Elementos de cerámica     NaN       NaN  2024-01-17   
139      2171    Parlantes Bluetooth     NaN       NaN  2024-01-17   

       categoria  
136   Decoración  
139  Electrónica

In [25]:
# Analizar los precios de los NaN a ver que opciones tenemos
df_ventas[df_ventas["producto"]=="Elementos de cerámica"]

id_venta               producto  precio  cantidad fecha_venta  \
36          17  Elementos de cerámica  108.48      11.0  2024-01-07   
51         687  Elementos de cerámica   27.68       8.0  2024-01-08   
71         817  Elementos de cerámica  106.31       1.0  2024-01-10   
95         507  Elementos de cerámica  111.10       1.0  2024-01-13   
97         987  Elementos de cerámica  102.62       3.0  2024-01-13   
...        ...                    ...     ...       ...         ...   
2892       427  Elementos de cerámica   60.13       6.0  2024-12-14   
2904       567  Elementos de cerámica   97.22       8.0  2024-12-15   
2961       917  Elementos de cerámica   52.65       8.0  2024-12-22   
2972       497  Elementos de cerámica  124.16       4.0  2024-12-23   
3011       167  Elementos de cerámica  104.96       6.0  2024-12-28   

       categoria  
36    Decoración  
51    Decoración  
71    Decoración  
95    Decoración  
97    Decoración  
...          ...  
2892  Decoración  
2904  Decoración  
2961  Decoración  
2972  Decoración  
3011  Decoración  

[100 rows x 6 columns]

In [26]:
df_ventas["producto"].value_counts()

producto
Lámpara de mesa           181
Auriculares               147
Microondas                135
Cafetera                  118
Laptop                    104
Secadora                  103
Jarrón decorativo         102
Smartphone                102
Alfombra                  102
SmartWatch                102
Aspiradora                102
Cortinas                  102
Rincón de plantas         102
Freidora eléctrica        101
Espejo decorativo         101
Adorno de pared           101
Lavadora                  100
Consola de videojuegos    100
Cuadro decorativo         100
Horno eléctrico           100
Plancha de vapor          100
Proyector                 100
Elementos de cerámica     100
Batidora                  100
Televisor                 100
Cámara digital            100
Parlantes Bluetooth        99
Tablet                     58
Heladera                   49
Candelabro                 24
Name: count, dtype: int64

### Observaciones




* Cantidad de registros: 3035
* Columnas: 6
* Algunos nulos (2), 1% del faltante, dropna
* Campo precio: encontre $, removerlo y
cambiar a float64
* Campo fecha: ajustar a dtype datetime

## Dataframe marketing

### Analisis exploratorio de datos

Aplicamos los métodos de EDA vistos en clase, `info`, `columns`, `dtypes`, `head`, `tail`, `sample`, `unique`, `value_counts`, `describe`, `duplicated`, `isnull`, etc. y documentamos las observaciones:

In [27]:
# Aplicamos info() para familiarizarnos con el dataframe
df_marketing.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90 entries, 0 to 89
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   id_campanha   90 non-null     int64         
 1   producto      90 non-null     object        
 2   canal         90 non-null     object        
 3   costo         90 non-null     float64       
 4   fecha_inicio  90 non-null     datetime64[ns]
 5   fecha_fin     90 non-null     datetime64[ns]
dtypes: datetime64[ns](2), float64(1), int64(1), object(2)
memory usage: 4.3+ KB


In [28]:
# Visualizamos los primeros registros con head()
# también podemos usar tail() para visualizar los últimos
df_marketing.head()

id_campanha         producto  canal  costo fecha_inicio  fecha_fin
0           74  Adorno de pared     TV   4.81   2024-03-20 2024-05-03
1           12           Tablet   RRSS   3.40   2024-03-26 2024-05-13
2           32  Lámpara de mesa  Email   5.54   2024-03-28 2024-04-20
3           21       Smartphone   RRSS   6.37   2024-03-29 2024-05-16
4           58         Alfombra  Email   4.25   2024-03-31 2024-05-05

In [29]:
# Analizamos el campo costo
df_marketing["costo"].sample(5)

23    5.24
22    4.53
69    4.59
13    4.20
81    4.19
Name: costo, dtype: float64

In [30]:
# Analizamos el campo fecha_inicio
df_marketing["fecha_inicio"].sample(5)

58   2024-09-08
47   2024-08-09
77   2024-11-03
12   2024-05-03
76   2024-11-03
Name: fecha_inicio, dtype: datetime64[ns]

In [31]:
# Analizamos el campo fecha_fin
df_marketing["fecha_fin"].sample(5)

18   2024-07-02
66   2024-10-24
74   2024-12-08
32   2024-09-03
27   2024-07-15
Name: fecha_fin, dtype: datetime64[ns]

### Calidad de los datos


In [71]:
# Buscamos registros duplicados
df_marketing.duplicated().sum()

np.int64(0)

### Observaciones
*** Completar ****

## Dataframe clientes

### Analisis exploratorio de datos

### Observaciones

# Etapa 2

## Limpieza de datos
Actividad 1 - Limpiar el conjunto de datos eliminando duplicados y caracteres no deseados. Documentar el proceso y los resultados.

### Dataframe ventas

* Duplicados: 35, deben ser eliminados -- Se eliminaron 35 duplicados  
* Nulos: 2, representan el 1% del faltante, dropna -- Se eliminaron 2 registros con 2 NaN.
* Campo precio: remover el simbolo $ y cambiar el dtype a float64
* Campo fecha: ajustar a dtype datetime
































In [32]:
# Eliminamos registros duplicados
# atributos a evaluar, keep, ignore_index, inplace
df_ventas.drop_duplicates(inplace=True, ignore_index=True)

In [33]:
# Validamos que no haya duplicados
df_ventas.duplicated().sum()

np.int64(0)

In [34]:
# Eliminamos registros con datos nulos
df_ventas.dropna(inplace=True)

In [35]:
# Validamos que no haya nulos
df_ventas.isnull().sum()

id_venta       0
producto       0
precio         0
cantidad       0
fecha_venta    0
categoria      0
dtype: int64

In [36]:
df_ventas.info()

<class 'pandas.core.frame.DataFrame'>
Index: 2998 entries, 0 to 2999
Data columns (total 6 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   id_venta     2998 non-null   int64         
 1   producto     2998 non-null   object        
 2   precio       2998 non-null   float64       
 3   cantidad     2998 non-null   float64       
 4   fecha_venta  2998 non-null   datetime64[ns]
 5   categoria    2998 non-null   object        
dtypes: datetime64[ns](1), float64(2), int64(1), object(2)
memory usage: 164.0+ KB


In [ ]:
# Eliminamos el caracter especial $ de precio
# Normalización: usar replace()
# De ser necesario, podemos usar el atributo regex=True
df_ventas["precio"] = df_ventas["precio"].str.replace("$", "", regex=False)

In [ ]:
# Validamos que precio no contenga $
df_ventas["precio"].sample(5)

In [38]:
# Convertimos precio de str a float
# Podemos usar pd.to_numeric o astype o incluso combinarlos
df_ventas["precio"] = df_ventas["precio"].astype(float)

In [39]:
# Convertir el dtype de cantidad, de float64 a int64
df_ventas["cantidad"] = df_ventas["cantidad"].astype(int)

In [40]:
# Validar la conversion a int
df_ventas["cantidad"].dtypes

dtype('int64')

In [41]:
# Convertir el campo fecha_venta a dtype datetime
# pd.to_datetime(df_ventas["fecha_venta"], format="%d/%m/%Y")
# pd.to_datetime(df_ventas["fecha_venta"], dayfirst=True)
df_ventas["fecha_venta"] = pd.to_datetime(df_ventas["fecha_venta"], format="%d/%m/%Y")

In [42]:
# Validar la conversion de fecha
df_ventas["fecha_venta"].sample(5)

345    2024-02-10
555    2024-03-06
2018   2024-08-28
140    2024-01-17
1903   2024-08-14
Name: fecha_venta, dtype: datetime64[ns]

In [43]:
df_ventas["fecha_venta"].dtype

dtype('<M8[ns]')

### Dataframe marketing

* fecha_inicio: debemos convertir el campo de object a dtype datatime
* fecha_fin: debemos convertir el campo de object a dtype datatime

In [44]:
df_marketing.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90 entries, 0 to 89
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   id_campanha   90 non-null     int64         
 1   producto      90 non-null     object        
 2   canal         90 non-null     object        
 3   costo         90 non-null     float64       
 4   fecha_inicio  90 non-null     datetime64[ns]
 5   fecha_fin     90 non-null     datetime64[ns]
dtypes: datetime64[ns](2), float64(1), int64(1), object(2)
memory usage: 4.3+ KB


In [45]:
# Convertir el campo fecha_inicio a dtype datetime
df_marketing["fecha_inicio"] = pd.to_datetime(df_marketing["fecha_inicio"], format="%d/%m/%Y")

In [46]:
# Convertir el campo fecha_fin a dtype datetime
df_marketing["fecha_fin"] = pd.to_datetime(df_marketing["fecha_fin"], format="%d/%m/%Y")

In [47]:
# Validamos los cambios
df_marketing.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90 entries, 0 to 89
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   id_campanha   90 non-null     int64         
 1   producto      90 non-null     object        
 2   canal         90 non-null     object        
 3   costo         90 non-null     float64       
 4   fecha_inicio  90 non-null     datetime64[ns]
 5   fecha_fin     90 non-null     datetime64[ns]
dtypes: datetime64[ns](2), float64(1), int64(1), object(2)
memory usage: 4.3+ KB


In [48]:
df_marketing["fecha_inicio"].dtype

dtype('<M8[ns]')

In [49]:
df_marketing["fecha_fin"].dtype

dtype('<M8[ns]')

## Transformación de datos
Actividad 2 - Aplicar filtros y transformaciones para crear una tabla de ventas que muestre solo los productos con alto rendimiento.

In [50]:
df_ventas.head(2)

id_venta           producto  precio  cantidad fecha_venta   categoria
0       792  Cuadro decorativo   69.94         5  2024-01-02  Decoración
1       811    Lámpara de mesa  105.10         5  2024-01-02  Decoración

In [51]:
# Transformación:
# Calculamos el valor de cada venta = precio * cantidad
# Agregamos una nueva columnas
df_ventas["valor_venta"] = df_ventas["precio"] * df_ventas["cantidad"]

In [52]:
# Visualizamos
df_ventas.head(2)

id_venta           producto  precio  cantidad fecha_venta   categoria  \
0       792  Cuadro decorativo   69.94         5  2024-01-02  Decoración   
1       811    Lámpara de mesa  105.10         5  2024-01-02  Decoración   

   valor_venta  
0        349.7  
1        525.5

In [53]:
# Agregación:
# Agrupar valor_venta por producto
# Sumamos las ventas dentro de cada grupo
tabla_ventas = (df_ventas.groupby("producto", as_index=False )["valor_venta"].sum())
tabla_ventas.head()

producto  valor_venta
0  Adorno de pared     48093.49
1         Alfombra     44773.06
2       Aspiradora     50085.86
3      Auriculares     74175.58
4         Batidora     50979.20

In [54]:
# Usemos describe para analizar algunos valores del dataframe resultante
tabla_ventas.describe()

valor_venta
count     30.000000
mean   48903.117333
std    13265.561715
min    11128.800000
25%    45601.920000
50%    48140.245000
75%    51092.960000
max    82276.380000

In [55]:
# Calcular el percentil 75 (umbral de alto rendimiento)
umbral_ventas = tabla_ventas["valor_venta"].quantile(0.75)
umbral_ventas

np.float64(51092.96)

In [56]:
# Identificar productos que superan el umbral
# productos_top = tabla_ventas.query("valor_venta >= @umbral")
productos_top = tabla_ventas[tabla_ventas["valor_venta"] >= umbral_ventas]
productos_top

producto  valor_venta
3         Auriculares     74175.58
5            Cafetera     59607.31
9   Cuadro decorativo     54297.60
16  Jarrón decorativo     51130.88
19    Lámpara de mesa     82276.38
20         Microondas     72562.89
25           Secadora     52115.45
27         Smartphone     54132.44

## Agregación
Actividad 3 - Resumir las ventas por categoría de producto y analizar los ingresos generados.

In [57]:
# Agrupamos por categoria y agregamos la suma
df_ventas.groupby("categoria", as_index=True)["valor_venta"].sum()

categoria
Decoración           479216.09
Electrodomésticos    505299.63
Electrónica          482577.80
Name: valor_venta, dtype: float64

In [58]:
# Renombrar nombre de la columna
ventas_categoria = df_ventas.groupby("categoria", as_index=False)["valor_venta"].sum()

In [59]:
# Ordenar y mostrar los resultados
ventas_categoria["valor_venta"].sort_values(ascending=False)

1    505299.63
2    482577.80
0    479216.09
Name: valor_venta, dtype: float64

## Integración de datos
Actividad 4 - Combinar los sets de datos de ventas y marketing para obtener una visión más amplia de las tendencias.

In [60]:
df_ventas.head(2)

id_venta           producto  precio  cantidad fecha_venta   categoria  \
0       792  Cuadro decorativo   69.94         5  2024-01-02  Decoración   
1       811    Lámpara de mesa  105.10         5  2024-01-02  Decoración   

   valor_venta  
0        349.7  
1        525.5

In [61]:
df_marketing.head(2)

id_campanha         producto canal  costo fecha_inicio  fecha_fin
0           74  Adorno de pared    TV   4.81   2024-03-20 2024-05-03
1           12           Tablet  RRSS   3.40   2024-03-26 2024-05-13

In [62]:
# Unir los datasets ventas y marketing por columna 'producto'
ventas_marketing = pd.merge(df_ventas, df_marketing, on="producto", how="left")

In [63]:
# Verificar coincidencias (muestra)
ventas_marketing.sample(5)

id_venta         producto  precio  cantidad fecha_venta  \
6033      2539        Proyector   33.87         7  2024-08-27   
1872      2858       SmartWatch   89.27         6  2024-03-13   
8658      1989         Batidora  124.17         7  2024-12-16   
3857      1616         Secadora   40.43         8  2024-05-26   
5014      1037  Horno eléctrico  120.58         8  2024-07-16   

              categoria  valor_venta  id_campanha  canal  costo fecha_inicio  \
6033        Electrónica       237.09           67     TV   3.68   2024-05-22   
1872        Electrónica       535.62           85     TV   5.07   2024-04-01   
8658  Electrodomésticos       869.19           57  Email   4.48   2024-04-10   
3857  Electrodomésticos       323.44           33  Email   3.80   2024-12-20   
5014  Electrodomésticos       964.64            5   RRSS   5.16   2024-08-22   

      fecha_fin  
6033 2024-07-12  
1872 2024-05-05  
8658 2024-06-08  
3857 2025-01-07  
5014 2024-10-17

In [64]:
ventas_marketing.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8994 entries, 0 to 8993
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   id_venta      8994 non-null   int64         
 1   producto      8994 non-null   object        
 2   precio        8994 non-null   float64       
 3   cantidad      8994 non-null   int64         
 4   fecha_venta   8994 non-null   datetime64[ns]
 5   categoria     8994 non-null   object        
 6   valor_venta   8994 non-null   float64       
 7   id_campanha   8994 non-null   int64         
 8   canal         8994 non-null   object        
 9   costo         8994 non-null   float64       
 10  fecha_inicio  8994 non-null   datetime64[ns]
 11  fecha_fin     8994 non-null   datetime64[ns]
dtypes: datetime64[ns](3), float64(3), int64(3), object(3)
memory usage: 843.3+ KB


In [65]:
ventas_marketing.groupby("categoria")[["valor_venta","costo"]].sum()

valor_venta     costo
categoria                               
Decoración          1437648.27  15774.86
Electrodomésticos   1515898.89  14644.65
Electrónica         1447733.40  14243.90

### Analizamos el impacto de la campaña

In [66]:
# Nos quedamos solo con los registros de ventas durante la campaña de marketing
# Seleccionamos solo los campos que nos interesan: "producto", "valor_venta", "costo"
ventas_durante_campanha = ventas_marketing[(ventas_marketing["fecha_venta"] >= ventas_marketing["fecha_inicio"]) & (ventas_marketing["fecha_venta"] <= ventas_marketing["fecha_fin"])]

In [67]:
# Agrupamos por producto y hacemos una agregación por "valor de venta" y "costo"
ventas_impacto = ventas_durante_campanha[["producto", "valor_venta", "costo"]]

In [68]:
ventas_impacto.head(5)

producto  valor_venta  costo
2073  Adorno de pared       698.52   4.81
2091  Adorno de pared       553.91   4.81
2124  Adorno de pared       579.54   4.81
2160  Adorno de pared       286.68   4.81
2256  Adorno de pared       831.20   4.81

In [ ]:
# Generamos una nueva columna con la diferencia entre lo vendido y lo invertido en la campaña
ventas_impacto["diferencia"] = ventas_impacto["valor_venta"] - ventas_impacto["costo"]

In [70]:
ventas_impacto.head(5)

producto  valor_venta  costo  diferencia
2073  Adorno de pared       698.52   4.81      693.71
2091  Adorno de pared       553.91   4.81      549.10
2124  Adorno de pared       579.54   4.81      574.73
2160  Adorno de pared       286.68   4.81      281.87
2256  Adorno de pared       831.20   4.81      826.39